### PR TITLE
[FEQ] george / FEQ-2641 / Implement Section Message Component 

### PR DESCRIPTION
**Pull request by heorhi-deriv**
- Implement `Section message` component
Screenshot 2024-09-05 at 18 11 16

- Fix border width for the Badge component:
The border width of the Badge component is being overridden by the Reset CSS file due to the CSS loading order in `deriv-app`. Adjust the CSS to ensure the `Badge` component's border width remains consistent.

### CODE REVIEW

File: src/components/SectionMessage/SectionMessage.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
+$error-bgColor: rgba(230, 25, 14, 0.08);
+$warning-bgColor: rgba(255, 156, 19, 0.08);
+$success-bgColor: rgba(0, 136, 50, 0.08);
+$info-bgColor: rgba(44, 154, 255, 0.08);
+$description-color: rgba(0, 0, 0, 0.72);
+$general-color: #f2f3f4;
```
Comment (aizad-deriv) [inline src/components/SectionMessage/SectionMessage.scss:6]: Isit possible to use the existing colors that we have in `themes.scss`

File: src/components/SectionMessage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+type SectionMessageProps = {
+    children: JSX.Element | string;
+    className?: string;
+    ctaSection?: JSX.Element;
+    title?: JSX.Element | string;
+    variant?: TVariant;
+};
+
+const VariantIcons: Record<Exclude<TVariant, "general">, JSX.Element> = {
```
Comment (aizad-deriv) [inline src/components/SectionMessage/index.tsx:22]:
```suggestion
const VariantIcons: Record<Exclude<TVariant, "general">, IconTypes> = {
```

There's a ready made type for the icons from `deriv-com/quill-icons`

`import { IconTypes } from '@deriv/quill-icons';`

File: src/components/SectionMessage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    error: <LabelPairedTriangleExclamationMdBoldIcon fill="#C40000" />,
+    info: <LabelPairedCircleInfoMdBoldIcon fill="#0777C4" />,
+    success: <LabelPairedCircleCheckMdBoldIcon fill="#007A22" />,
+    warning: <LabelPairedCircleExclamationMdBoldIcon fill="#C47D00" />,
```
Comment (aizad-deriv) [inline src/components/SectionMessage/index.tsx:26]: I believe you can use the css variables instead of hex code here aswell

**Review by shayan-deriv — APPROVED** [review]

File: src/components/SectionMessage/SectionMessage.scss (hunk 1 above)
Comment (heorhi-deriv) [reply src/components/SectionMessage/SectionMessage.scss:6]: sorted

File: src/components/SectionMessage/index.tsx (hunk 2 above)
Comment (heorhi-deriv) [reply src/components/SectionMessage/index.tsx]: sorted

File: src/components/SectionMessage/index.tsx (hunk 3 above)
Comment (heorhi-deriv) [reply src/components/SectionMessage/index.tsx]: sorted

File: src/components/SectionMessage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    variant = "general",
+}: SectionMessageProps) => (
+    <div
+        className={clsx(
+            "deriv-section-message",
+            VariantClasses[variant],
+            className,
+        )}
+    >
+        {variant !== "general" && (
```
[the remark this answers is not in the document]
Comment (heorhi-deriv) [reply src/components/SectionMessage/index.tsx]: @wojciech-deriv added possibility to pass custom icon

**Review by jim-deriv — APPROVED** [review]

**Review by shayan-deriv — APPROVED** [review]

**Pull request merged**